rs/ledger_suite/icrc1/ledger/src/main.rs

```diff
@@ -164,43 +184,31 @@ fn post_upgrade(args: Option<LedgerArgument>) {
     let mut pre_upgrade_instructions_consumed = 0;
 
     if !memory_manager_found {
-        let mut stable_reader = StableReader::default();
-        LEDGER.with(|cell| {
-            *cell.borrow_mut() = Some(
-                ciborium::de::from_reader(&mut stable_reader)
-                    .expect("failed to decode ledger state"),
-            );
-        });
+        let msg =
+            "Cannot upgrade from scratch stable memory, please upgrade to memory manager first.";
+        ic_cdk::println!("{msg}");
+        panic!("{msg}");
```

Comment:
Do one or both of these appear in kibana if this happens on mainnet?

Reply:
Probably both, but, as discussed offline, since this ends with a `panic`, it will not pollute the logs too much.